### PR TITLE
Print linerror

### PR DESCRIPTION
**Pull request by latot**
Maybe if you like, set a `sympref debug 1`?

Fixes https://github.com/cbm755/octsympy/issues/636

Thx. Cya.

### CODE REVIEW

**Comment by latot** [comment]
mm, this for a octave bug, is fixed in the actual hg version.

**Comment by cbm755** [comment]
Seems like a good idea.  Please do `strtrim (cmd{A{3}})));`  (i.e., some whitespace problems in the push).

I don't think its an Octave bug: I think its just that on other ipc modes, there are some lines added so you cannot look look it up at index `A{3}`.  Does it work if you use `errlineno` instead?

**Comment by latot** [comment]
Hi, i'm already using ```strtrim```, so should be fine, and you are right, don't is a octave bug, i was testing this bad (i was running python_cmd tests instead of sympref), in popen2 it works in some cases (coincidences), so now works.

Thx. Cya.

File: inst/python_cmd.m [hunk 1]
```diff
@@ -170,8 +170,9 @@
   %% Error reporting
   % ipc drivers are supposed to give back these specially formatting error strings
   if (~isempty(A) && ischar(A{1}) && strcmp(A{1}, 'COMMAND_ERROR_PYTHON'))
-    errlineno = A{3} - db.prelines - LinesBeforeCmdBlock;
-    error(sprintf('Python exception: %s\n    occurred at line %d of the Python code block', A{2}, errlineno));
+    errcmdlineno = A{3} - db.prelines;
+    errlineno = errcmdlineno - LinesBeforeCmdBlock;
+    error(sprintf('Python exception: %s\n    occurred at line %d of the Python code block:\n    %s', A{2}, errlineno, strtrim (cmd {errcmdlineno})));
```
Comment (cbm755) [inline inst/python_cmd.m]: my comment was about whitespace `cmd{...}` (no spaces with indexing)

**Comment by latot** [comment]
!! Okis, now i fixed it.

**Comment by latot** [comment]
Travis timeout, but in Python3 (stalled) pass sympref with the relevant tests.

**Comment by cbm755** [comment]
I restarted the CI run.

Probably this should have a test or two.  IIRC, there are tests for such things in `sympref`.

**Comment by latot** [comment]
Okidokis, Tests added!

**Comment by cbm755** [comment]
But these tests do not test the new feature!

I'm trying to see why we need both `errcmdlineno` and `errlineno`, why they are not the same.  Figured maybe a test would help.

Also some commit should have a "Fixes #636" in the commit message.

**Comment by latot** [comment]
Hi, i think the test are fine, because the idea of this is send what line has the error:
```
python_cmd('pass;raise;pass')
```
this should return ```raise``` and some other text, this is what test do.

```errcmdlineno``` don't are equal to ```errlineno``` because, the last one try found the error counting from the input data of python_cmd (```def _fcn(_ins):```), so we need remove some lines and remove the start lines of the default cmd, while in the first, we are trying to get the error not in the input of python_cmd, is in the complete cmd. I think i'm explaining this too bad, so lets use an example:

```
cmd = [{'Hi'
             'O'}
             some user input text
            {'Bye'
             'J'}]
```
well, if i send this text, ```{'pass' ; 'Meh' ; 'pass'}```, in what position is ```Meh```?, is the line number 2, even if is replaced in cmd, should be 4(input text) - 2 (pre lines), this is errlineno, but in what line of cmd is?, here is 4, and this is the errcmdlineno, and as you can see both are related, basically this is.

**Comment by cbm755** [comment]
Thanks, this looks good to me.  I want to add a similar test to the system ipc tests, but I can do that.

**Comment by cbm755** [comment]
I added 6c61547366e1fbce8be20d7bf048737f4246ec48 just so it gets tested with `system` as well.

**Pull request merged**